public/app/features/profile/ChangePassword.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { PureComponent } from 'react';
+import { hot } from 'react-hot-loader';
+import { connect } from 'react-redux';
+import { getBackendSrv } from '@grafana/runtime';
+import config from 'app/core/config';
+import { StoreState } from 'app/types';
+import { getNavModel } from 'app/core/selectors/navModel';
+import { NavModel } from '@grafana/data';
+import { Button, LinkButton } from '@grafana/ui';
+import { UserProvider } from 'app/core/utils/UserProvider';
+import Page from 'app/core/components/Page/Page';
+import { PasswordInput } from 'app/core/components/PasswordInput/PasswordInput';
+
+export interface Props {
+  navModel: NavModel;
+}
+
+export interface State {
+  oldPassword: string;
+  newPassword: string;
+  confirmNew: string;
+}
+
+export class ChangePassword extends PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      oldPassword: '',
+      newPassword: '',
+      confirmNew: '',
+    };
+  }
+
+  onOldPasswordChange(oldPassword: string) {
+    this.setState({ oldPassword });
+  }
+
+  onNewPasswordChange(newPassword: string) {
+    this.setState({ newPassword });
+  }
+
+  onConfirmPasswordChange(confirmNew: string) {
+    this.setState({ confirmNew });
+  }
+
+  onChangePassword = async () => {
```

Comment:
You don't need that, right?

Reply:
No. I did it the previous way first - before attempting the provider. Will remove after I get provider bit correct.